File: JavaScript_Courses/JavaScript_Basics/Preparation/05_While_Loop_Exercise/03_Vacation.js
function vacation(input) {
    let moneyNeeded = Number(input[0]);
    let moneyReal = Number(input[1]);

    let i = 2;

    let consecutiveSpendingDays = 0;
    let totalDays = 0;

    while (true) {
        let currentAction = input[i];
        let currentAmount = Number(input[i + 1]);

        totalDays++;

        if (currentAction == "spend") {
            consecutiveSpendingDays++;
            moneyReal -= currentAmount;
            if (moneyReal < 0) {
                moneyReal = 0;
            }
        } else if (currentAction == "save") {
            moneyReal += currentAmount;
            consecutiveSpendingDays = 0;
        }

        if (consecutiveSpendingDays == 5) {
            console.log("You can't save the money.");
            console.log(totalDays);
            break;
        }

        if (moneyReal >= moneyNeeded) {
            console.log(`You saved the money for ${totalDays} days.`);
            break;
        }

        i += 2;
    }
}


// vacation([
//     "2000",
//     "1000",
//     "spend",
//     "1200",
//     "save",
//     "2000"
// ]);

File: Python_Courses/Python_Basics/Python_PB_2023_04/05_While_Loop_Exercise/03_Vacation.py
vacation_cost = float(input())
starting_money = float(input())

total_days = 0
consecutive_spend_days = 0

total_money = starting_money
are_money_enough = False
while True:
    action = input()  # "spend" или "save"
    current_amount = float(input())
    total_days += 1

    if action == "spend":
        consecutive_spend_days += 1
        total_money -= current_amount
        if total_money < 0:
            total_money = 0

        if consecutive_spend_days >= 5:
            break

    elif action == "save":
        consecutive_spend_days = 0
        total_money += current_amount
        if total_money >= vacation_cost:
            are_money_enough = True
            break


if are_money_enough:
    print(f"You saved the money for {total_days} days.")
else:
    print("You can't save the money.")
    print(total_days)
